icinga-app/icinga.cpp

```diff
@@ -906,24 +906,14 @@ int main(int argc, char **argv)
 #ifndef _WIN32
 	String keepFDs = Utility::GetFromEnvironment("ICINGA2_KEEP_FDS");
 	if (keepFDs.IsEmpty()) {
-		rlimit rl;
-		if (getrlimit(RLIMIT_NOFILE, &rl) >= 0) {
-			rlim_t maxfds = rl.rlim_max;
-
-			if (maxfds == RLIM_INFINITY)
-				maxfds = 65536;
-
-			for (rlim_t i = 3; i < maxfds; i++) {
-				int rc = close(i);
-
+		Utility::CloseAllFDs<3>(
+			{0, 1, 2}
 #ifdef I2_DEBUG
```

Comment:
I'm not a huge fan of having this ifdef within the function call. I'd either move it to within the lambda, so that it reduces to `[](int fd){}` if it's not defined, or set a variable for the callback within an ifdef.